femwell/maxwell/waveguide.py

```diff
@@ -280,9 +282,119 @@ def plot(self, field, plot_vectors=False, colorbar=True, direction="y", title="E
+
+        if ax is None:
+            fig, ax = plt.subplots()
+        else:
+            fig = plt.gcf()
+
+        if field == "E":
+            mfield = self.E
+        elif field == "H":
+            mfield = self.H
+        else:
+            raise ValueError("A valid field is 'E' or 'H'.")
+
+        (mfield_t, mfield_t_basis), (mfield_n, mfield_n_basis) = self.basis.split(mfield)
+
+        if component == "x" or component == "y":
+            plot_basis = mfield_t_basis.with_element(ElementVector(ElementDG(ElementTriP1())))
+            mfield_xy = plot_basis.project(mfield_t_basis.interpolate(conv_func(mfield_t)))
+            (mfield_x, mfield_x_basis), (mfield_y, mfield_y_basis) = plot_basis.split(mfield_xy)
+            if component == "x":
+                mfield_x_basis.plot(mfield_x, ax=ax, shading="gouraud")
+            else:
+                mfield_y_basis.plot(mfield_y, ax=ax, shading="gouraud")
+        elif component == "t":
+            plot_basis = mfield_t_basis
+            mfield_t_basis.plot(conv_func(mfield_t), ax=ax, shading="gouraud")
+        elif component == "z" or component == "n":
+            plot_basis = mfield_n_basis
+            mfield_z, mfield_z_basis = mfield_n, mfield_n_basis
+            mfield_z_basis.plot(conv_func(mfield_z), ax=ax, shading="gouraud")
+        else:
+            raise ValueError("A valid component is 'x', 'y', 'z', 'n' or 't'.")
+
+        if boundaries:
+            for subdomain in plot_basis.mesh.subdomains.keys() - {"gmsh:bounding_entities"}:
+                plot_basis.mesh.restrict(subdomain).draw(ax=ax, boundaries_only=True, color="k")
+        if colorbar:
+            divider = make_axes_locatable(ax)
+            cax = divider.append_axes("right", size="5%", pad=0.05)
+            plt.colorbar(ax.collections[-1], cax=cax)
+
+        ax.set_title(f"{field}{component} ({part}. part)")
+
+        return fig, ax
+
+    # def show(
+    #     self,
+    #     field: Literal["E", "H"] ,
+    #     part: Literal["real", "imag", "abs"] = "real",
+    #     boundaries: bool = True,
+    #     colorbar: bool = False,
+    # ):
+    #     fig, axs = plt.subplots(1, 3, subplot_kw=dict(aspect=1))
+
+    #     for id_ax, comp in enumerate("xyz"):
+    #         self.plot_component(field, comp, part, boundaries, colorbar, axs[id_ax])
+    #     plt.tight_layout()
+    #     plt.show()
+
+    def show(self, field: Literal["E", "H"] | NDArray, **kwargs):
+        if type(field) is np.ndarray:
+            warn(
+                "The behavior of passing an array directly to `show` "
+                + "is deprecated and will be removed in the future. "
+                + "Use `plot` instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            self.plot(field=field, **kwargs)
+            plt.show()
+        else:
+            from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+            part = kwargs.get("part", "real")
```

Comment:
would there be a way to include those in the signature of the function directly instead of using kwargs?

It seems to me that `part` would be the only variable which is not used by the old function? (There could just be a default value for part which is ignored with a warning in case the old function is used?

Reply:
Yeah, I agree removing the `**kwargs` is a better practice :)
For `part`, it should not be checked if `show` is used in the old way, so simply giving it a default value should be enough!